File: chaeryeon823/BOJ_11399.js
// ATM
// 정렬
// S4

const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let inputCount = 0;
let n;
let arr;

rl.on("line", (line) => {
  if (inputCount === 0) {
    n = parseInt(line);
    inputCount++;
  } else {
    arr = line.split(" ").map(Number);
    arr.sort((a, b) => a - b);

    let result = 0;
    let j = 0;

    for (let i = n; i > 0; i--) {
      result += arr[j] * i;
      j++;
    }

    console.log(result);
    rl.close();
  }
});

File: youbbin/BOJ_11399.py
import sys

n = int(sys.stdin.readline())
p = list(map(int, sys.stdin.readline().split()))

p.sort() # 오름차순

total = 0
withdraw = 0

for time in p:
    withdraw += time  # 한 사람당 인출시간
    total += withdraw # 총합

# 4. 결과 출력
print(total)